synapse/storage/databases/main/event_push_actions.py

```diff
@@ -57,6 +57,13 @@ class EmailPushAction(HttpPushAction):
     received_ts: Optional[int]
 
 
+@attr.s(slots=True, frozen=True, auto_attribs=True)
+class UserPushAction(EmailPushAction):
+    topological_ordering: int
+    highlight: bool
+    profile_tag: str
```

Comment:
I'm struggling to understand what this actually represents, and why it's different to the other types of PushAction. If you've managed to grok it, maybe you could add some docstring?

Reply:
It seems to be required by the spec, for what it is worth: https://spec.matrix.org/latest/client-server-api/#get_matrixclientv3notifications

It is defined there as:

> The profile tag of the rule that matched this event.

I can add something along those lines if it is useful, but I'm not sure it really is. It might make more sense to say that these applies to the `/notifications` endpoint somewhere? But it applies to a store method so it gets a bit convoluted?

Reply:
In case it was unclear: by "this" I meant the entire `UserPushAction` class.

If `UserPushAction` corresponds to something in the spec, it would probably be worth saying that.

Reply:
Ah, fair! I thought you meant `profile_tag` in particular. :)

Reply:
I'll add some docstrings to the three classes then. 👍